app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt

```diff
@@ -0,0 +1,188 @@
+    FlexibleAdapter.OnItemClickListener,
+    FlexibleAdapter.OnUpdateListener,
+    FabController {
+
+    private var recycler: RecyclerView? = null
+    private var adapter: FlexibleAdapter<ClearDatabaseSourceItem>? = null
+
+    private var menu: Menu? = null
+
+    var actionFab: ExtendedFloatingActionButton? = null
+    var actionFabScrollListener: RecyclerView.OnScrollListener? = null
+
+    private val selectedSources: List<ClearDatabaseSourceItem>
+        get() = adapter?.selectedPositions?.mapNotNull(adapter!!::getItem) ?: emptyList()
+
+    init {
+        setHasOptionsMenu(true)
+    }
+
+    override fun createBinding(inflater: LayoutInflater): ClearDatabaseControllerBinding {
+        return ClearDatabaseControllerBinding.inflate(inflater)
+    }
+
+    override fun createPresenter(): ClearDatabasePresenter {
+        return ClearDatabasePresenter()
+    }
+
+    override fun getTitle(): String? {
+        return activity?.getString(R.string.pref_clear_database)
+    }
+
+    override fun onViewCreated(view: View) {
+        super.onViewCreated(view)
+
+        binding.recycler.applyInsetter {
+            type(navigationBars = true) {
+                padding()
+            }
+        }
+
+        adapter = FlexibleAdapter<ClearDatabaseSourceItem>(null, this, true)
+        binding.recycler.adapter = adapter
+        binding.recycler.layoutManager = LinearLayoutManager(activity)
+        binding.recycler.setHasFixedSize(true)
+        adapter?.fastScroller = binding.fastScroller
+        recycler = binding.recycler
+    }
+
+    override fun onDestroyView(view: View) {
+        adapter = null
+        super.onDestroyView(view)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.generic_selection, menu)
+        this.menu = menu
+        menu.apply {
+            get(0).isVisible = adapter?.itemCount ?: 0 > 0
+            get(1).isVisible = adapter?.itemCount ?: 0 > 0
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val adapter = adapter ?: return false
+        when (item.itemId) {
+            R.id.action_select_all -> {
+                for (i in 0..adapter.itemCount) {
+                    addSelection(i)
+                }
+            }
+            R.id.action_select_inverse -> {
+                for (i in 0..adapter.itemCount) {
+                    toggleSelection(i)
+                }
+            }
+        }
+        adapter.notifyItemRangeChanged(0, adapter.itemCount, Payload.SELECTION)
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onUpdateEmptyView(size: Int) {
+        if (size > 0) {
+            binding.emptyView.hide()
+        } else {
+            binding.emptyView.show(activity!!.getString(R.string.database_clean))
+        }
+
+        menu?.apply {
+            get(0).isVisible = size > 0
+            get(1).isVisible = size > 0
+        }
```

Comment:
Can just loop over them to change the visibility accordingly
```suggestion
        menu?.forEach { menuItem -> menuItem.isVisible = size > 0 }
```